app/gui/src/dashboard/providers/DriveProvider.tsx

```diff
@@ -259,20 +259,25 @@ export function useSetVisuallySelectedKeys() {
 /** Toggle whether a specific directory is expanded. */
 export function useToggleDirectoryExpansion() {
   const driveStore = useDriveStore()
-  const setExpandedDirectoryIds = useSetExpandedDirectoryIds()
+  const { category } = useCategoriesAPI()
+  const setExpandedDirectories = useSetExpandedDirectories()
 
-  return useEventCallback((directoryId: DirectoryId, override?: boolean) => {
-    const expandedDirectoryIds = driveStore.getState().expandedDirectoryIds
-    const isExpanded = expandedDirectoryIds.includes(directoryId)
+  return useEventCallback((id: DirectoryId, override?: boolean) => {
+    const expandedDirectories = driveStore.getState().expandedDirectories
+    const isExpanded = expandedDirectories[category.rootPath]?.includes(id) ?? false
```

Comment:
good candidate for encapsulation

Reply:
done (locally, will push soon :tm:) if you mean replacing all `setExpandedDirectories` with this